lesson11/main.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+const endLoading = () => wrapper.remove();
+const jsonURL = "https://jsondata.okiba.me/v1/json/KKl45211103113313";
+
+async function fetchData() {
+   startLoading();
```

Comment:
IMO

I think `startLoading` and `endLoading` are not part of fetchData because that name means just like a getting data.

So if I were you I would run them in the function `runCreateListElement` and also change the function name from `runCreateListElement` to `init` or something :)

Like this ↓
```
async function init() {
 startLoading();
   try {
      const result = await fetchData();
      createListElements(result);
   } catch {
      alert(`データが取得できませんでした`)
      console.error('エラー');
   }
 endLoading();
};
```

Reply:
Thanks for your review.
I now understand that it is better to run it in the place as you pointed out.

I was wondered what kind of name I should name.
Is “init" often given to the first thing to be run?

c8e776b

Reply:
Yeah, it's a little hard to think of a name for the function. I always take long time to name them...🤣 
`init` stands for initialize or initialization, so I think so.
For me, it's like a minimum setup that I have to do in order to do the requirements (in this case, to do assignment 11).
I'm not sure it will help, but I'll share this website.
[「分かりそう」で「分からない」でも「分かった」気になれるIT用語辞典](https://wa3.i-3-i.info/word12961.html)

In this case, I used `init` because it includes some loading processes, but if I didn't include them in this function, I think `runCreateListElement` should be fine.

Reply:
Thank you for explanation. I read the URL link you gave me.
I'm going to try to write readable code like you do.